onnxruntime/core/providers/cpu/ml/linearregressor.h

```diff
@@ -15,12 +16,50 @@ template <typename T>
 class LinearRegressor final : public OpKernel {
  public:
   LinearRegressor(const OpKernelInfo& info);
-  Status Compute(OpKernelContext* context) const override;
+
+  Status Compute(OpKernelContext* ctx) const override {
+    concurrency::ThreadPool* tp = ctx->GetOperatorThreadPool();
+
+    const auto* X = ctx->Input<Tensor>(0);
+    if (X->Shape().NumDimensions() == 0) {
+      return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
+                    "Input shape needs to be at least a single dimension.");
+    }
+    // X: [N, feature_size],coefficients_[target, feature_size]
+    // X*coefficients_^t : [N, target]
+    int64_t stride = X->Shape().NumDimensions() == 1 ? X->Shape()[0] : X->Shape()[1];
+    int64_t N = X->Shape().NumDimensions() == 1 ? 1 : X->Shape()[0];
+    Tensor* Y = ctx->Output(0, TensorShape({N, targets_}));
+    T* Ydata = Y->MutableData<T>();
+    //math::Gemm doesn't support double in all the platforms, so we use MatMul here.
+    math::MatMul<T>(static_cast<int>(N), static_cast<int>(targets_), static_cast<int>(stride), X->Data<T>(),
+                    coefficients_.data(), Y->MutableData<T>(), tp);
+    bool useIntercepts = intercepts_.size() == static_cast<size_t>(targets_);
+    if (useIntercepts) {
+      for (int64_t i = 0; i < N; i++)  // for each point
+      {
```

Comment:
Can you use ParallelFor for this or is it not worth it?

Reply:
Which ParallelFor ?  ParallelFor from the threadpool or eigen device?

Reply:
I was thinking of the ThreadPool method but whatever works best. Not required for this PR.

---
In reply to: [336697105](https://github.com/microsoft/onnxruntime/pull/1959#discussion_r336697105) [](ancestors = 336697105)